test/WebJobs.Extensions.Tests/Extensions/DocumentDB/Trigger/CosmosDBTriggerAttributeBindingProviderTests.cs

```diff
@@ -21,18 +22,22 @@ public class CosmosDBTriggerAttributeBindingProviderTests
         [MemberData("InvalidCosmosDBTriggerParameters")]
         public async Task InvalidParameters_Fail(ParameterInfo parameter)
         {
-            CosmosDBTriggerAttributeBindingProvider provider = new CosmosDBTriggerAttributeBindingProvider("notAConnectionString", string.Empty);
-            
+            CosmosDBTriggerAttributeBindingProvider provider = new CosmosDBTriggerAttributeBindingProvider(new DefaultNameResolver(), "notAConnectionString", string.Empty);
+
             InvalidOperationException ex = await Assert.ThrowsAsync<InvalidOperationException>(() => provider.TryCreateAsync(new TriggerBindingProviderContext(parameter, CancellationToken.None)));
 
-            Assert.NotNull(ex);            
+            Assert.NotNull(ex);
         }
 
         [Theory]
         [MemberData("ValidCosmosDBTriggerParameters")]
         public async Task ValidParameters_Succeed(ParameterInfo parameter)
         {
-            CosmosDBTriggerAttributeBindingProvider provider = new CosmosDBTriggerAttributeBindingProvider("AccountEndpoint=https://someEndpoint;AccountKey=someKey;", string.Empty);
+            var nameResolver = new TestNameResolver();
+            nameResolver.Values[DocumentDBConfiguration.AzureWebJobsDocumentDBConnectionStringName] = "AccountEndpoint=https://someEndpoint;AccountKey=someKey;";
+            nameResolver.Values["CosmosDBConnectionString"] = "AccountEndpoint =https://someEndpoint;AccountKey=someKey;";
+
+            CosmosDBTriggerAttributeBindingProvider provider = new CosmosDBTriggerAttributeBindingProvider(nameResolver, "AccountEndpoint=https://someEndpoint;AccountKey=someKey;", string.Empty);
```

Comment:
In my previous comments, I suggested reducing this down to just take an `INameResolver` and the `ChangeFeedHostOptions` as well as exposing some extra properties on your Binding for testing. If you do that, now you should be able to:

1. Give each value in your `TestNameResolver` a unique connection string/ key value.
2. Look at the exposed `DocumentCollectionLocation` (and `LeaseCollectionLocation`) to make sure that they have the correct values.

You may have to split up your test since the validation will vary based on the scenario -- or you can add 'expected' parameters in your MemberData, which we do elsewhere as well.

But this way you'll know that you properly set the right fallback values based on what the attribute asked for.